llmfoundry/tokenizers/tiktoken.py

```diff
@@ -50,6 +50,23 @@ def __init__(self,
             raise ImportError(
                 'You need to install tiktoken to use TiktokenTokenizerWrapper.')
 
+        # Workaround to make tiktokenizer picklable.
+        # https://github.com/huggingface/datasets/issues/5536#issuecomment-1682309347
+        # There is an open PR from HF to add this to tiktoken: https://github.com/openai/tiktoken/pull/181
+        import copyreg
+        import functools
+
+        from tiktoken import Encoding  # type: ignore (thirdParty)
```

Comment:
doing tiktoken.Encoding does not work and I cannot find a way to satisfy pyright that way, so I changed this to import Encoding directly. I think it's related to this (https://github.com/microsoft/pyright/issues/4863 ), but I don't totally understand what pyright can model and cannot model w.r.t. to imports

Reply:
type ignore on the line that used tiktoken.Encoding did not work?

Reply:
nope, I tried that, but it was erroring on a line that did not even use tiktoken.Encoding, and even adding type ignore to that line didn't work...

Reply:
https://github.com/mosaicml/llm-foundry/actions/runs/6699130750/job/18202680494 this was the failure:
`home/runner/work/llm-foundry/llm-foundry/llmfoundry/tokenizers/tiktoken.py:70:33 - error: "Encoding" is not a known member of module`, L70 doesn't even contain any ref to tiktoken.Encoding and adding a type ignore there didn't work

Reply:
ok, sure 🤷‍♂️ im not gonna think about this too hard. whatever works

Reply:
haha yeah, well this way we only need one type ignore also 😄